src/worker.jl

```diff
@@ -229,20 +299,46 @@ function worker_init(f::File)
                         )
                         continue
                     end
+                    # See whether the current MIME type needs to be handled
+                    # specially and embedded in a raw markdown block and whether
+                    # we should skip attempting to render any other MIME types
+                    # to may match.
+                    skip_other_mimes, new_mime, new_buffer = _transform_output(mime, buffer)
                     # Only send back the bytes, we do the processing of the
                     # data on the parent process where we have access to
                     # whatever packages we need, e.g. working out the size
                     # of a PNG image or converting a JSON string to an
                     # actual JSON object that avoids double serializing it
                     # in the notebook output.
-                    result[mime] = (; error = false, data = take!(buffer))
+                    result[new_mime] = (; error = false, data = take!(new_buffer))
+                    skip_other_mimes && break
                 end
             end
             return result
         end
-        render_mimetypes(value::Nothing) =
+        render_mimetypes(value::Nothing, cell_options) =
             Dict{String,@NamedTuple{error::Bool, data::Vector{UInt8}}}()
 
+        # Our custom MIME types need special handling. They get rendered to
+        # `text/markdown` blocks with the original content wrapped in a raw
+        # markdown block. MIMEs that don't match just get passed through.
+        function _transform_output(mime::String, buffer::IO)
+            mapping = Dict(
+                "QuartoNotebookRunner/openxml" => (true, "text/markdown", "openxml"),
+                "QuartoNotebookRunner/typst" => (true, "text/markdown", "typst"),
```

Comment:
> Will only the first matching MIME type get rendered when we know the output format?

Depends on whether these are `true` or `false`. Might be sufficient control of what gets rendered.